opal/core/patient_lists.py

```diff
@@ -119,6 +119,69 @@ def get_template_prefixes(self):
         return possible
 
 
+"""
+Sometimes we group lists for display purposes.
+"""
+class TabbedPatientListGroup(discoverable.DiscoverableFeature):
+    """
+    Groups of Patient Lists to display as tabs at the top of
+    any list in the group
+    """
+    module_name   = 'patient_lists'
+    member_lists  = []
+
+    @classmethod
+    def for_list(klass, patient_list):
+        """
+        Returns the group for a given PatientList.
+        Raises ValueError if not passed a PatientList
+        """
+        msg = 'TabbedPatientListGroup.for_list must be passed a PatientList'
+        try:
+            if not issubclass(patient_list, PatientList):
+                raise ValueError(msg)
+        except TypeError:
+                raise ValueError(msg)
+
+        for group in klass.list():
+            if patient_list in group.get_member_lists():
+                return group
+
+    @classmethod
+    def get_member_lists(klass):
+        """
+        A hook for dynamically customising the members of this list group.
+
+        Returns an iterable of PatientLists
+        Defaults to the `.member_lists` property
+        """
+        for l in klass.member_lists:
+            yield l
```

Comment:
> should we add user as an optional argument here?

And collapse the two methods `get_member_lists` and `get_member_lists_for_user` into one presumably? 
You could obv... I'm not sure what it would get you TBH ?

I guess my argument would be that it gives you slightly clearer hooks to override in your implementation - this method just gets members, and doesn't care about permissions / visibility / who, this other method cares about determining which of the members can be seen by this person.

Not sure it makes a huge amount of difference either way...

> change klass.member_lists to be a tuple 

Not exactly sure what you're getting at here? 

The original implementation of this was to just return `klass.member_lists`... 
I changed it to be a generator for some sense of internal API consistency once the class had grown `get_member_lists_for_user` which iterated over things and returns a generator.

In general I don't have particularly strong feelings about returning lists vs. generators, I tend towards implementing generators and letting calling code cast to a list if it's going to iterate through multiple times for some reason.